operators/src/processing/raster_kernel/mod.rs

```diff
@@ -0,0 +1,737 @@
+}
+
+#[derive(Debug, Clone)]
+enum InitializedRasterKernelMethod {
+    Convolution { matrix: Grid2D<f64> },
+    StandardDeviation { dimensions: GridShape2D },
+}
+
+impl TryFrom<RasterKernelMethod> for InitializedRasterKernelMethod {
+    type Error = RasterKernelError;
+
+    fn try_from(kernel_method: RasterKernelMethod) -> Result<Self, Self::Error> {
+        let dimensions = kernel_method.kernel_size();
+
+        ensure!(dimensions.number_of_elements() > 0, error::DimensionsZero);
+        ensure!(
+            dimensions.axis_size_x().is_odd() && dimensions.axis_size_y().is_odd(),
+            error::DimensionsNotOdd { actual: dimensions }
+        );
+
+        match kernel_method {
+            RasterKernelMethod::Convolution { matrix } => {
+                let matrix = Grid2D::new(dimensions, matrix.into_iter().flatten().collect())
+                    .map_err(|_| RasterKernelError::MatrixNotRectangular)?;
+
+                Ok(Self::Convolution { matrix })
+            }
+            RasterKernelMethod::StandardDeviation { .. } => {
+                Ok(Self::StandardDeviation { dimensions })
+            }
+        }
+    }
+}
+
+impl RasterKernelMethod {
+    fn kernel_size(&self) -> GridShape2D {
+        match self {
+            Self::Convolution { matrix } => {
+                let x_size = matrix.len();
+                let y_size = matrix.get(0).map_or(0, Vec::len);
+                GridShape2D::new([x_size, y_size])
+            }
+            Self::StandardDeviation { dimensions } => GridShape2D::new(*dimensions),
+        }
+    }
+}
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)), context(suffix(false)), module(error))]
+pub enum RasterKernelError {
+    #[snafu(display(
+        "The maximum kernel size is {}x{}, but the kernel has size {}x{}",
+        limit.axis_size_x(), limit.axis_size_y(), actual.axis_size_x(), actual.axis_size_y()
+    ))]
+    TooLarge {
+        limit: GridShape2D,
+        actual: GridShape2D,
+    },
+
+    #[snafu(display(
+        "The kernel must have an odd number of rows and columns, but the kernel has size {}x{}",
+        actual.axis_size_x(), actual.axis_size_y()
+    ))]
+    DimensionsNotOdd { actual: GridShape2D },
+
+    #[snafu(display("The kernel matrix dimensions must not be 0"))]
+    DimensionsZero,
+
+    #[snafu(display("The kernel matrix must be rectangular"))]
+    MatrixNotRectangular,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for RasterKernel {
+    async fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let tiling_specification = context.tiling_specification();
+
+        // dimensions must not be larger as the neighboring tiles
+        let dimensions = self.params.kernel.kernel_size();
+        ensure!(
+            dimensions.axis_size_x() <= tiling_specification.tile_size_in_pixels.axis_size_x()
```

Comment:
tile_size_in_pixels + 1 would be ok too :D

Reply:
Sure, but it is a reasonable maximum